src/example/mod.rs

```diff
@@ -20,16 +20,31 @@ pub mod prelude {
     pub use example::parser4::*;
 }
 
+/// An ID of a parser for the `peel_example` parsers.
+#[derive(Clone, Debug, Eq, PartialEq, Ord, PartialOrd, Hash)]
+pub enum ParserId {
+    /// `parser1::Parser1`
+    Parser1,
+    /// `parser2::Parser2`
+    Parser2,
+    /// `parser3::Parser3`
+    Parser3,
+    /// `parser4::Parser4`
+    Parser4,
+    /// Something else
+    Other(usize),
```

Comment:
What is the intention of `Other` here? How could a use case look like?

Reply:
It's used in one of the examples, where there's another fifth parser added.

I'm not entirely sure this is the right way. I think the approach with enum works well if the whole parser tree is provided by a library and not extended, or when the user builds his own tree (and uses his own enum). But the hybrid approach doesn't seem nice. On the other hand, I didn't want to kill the example altogether, but maybe in the long term, the example could be extracted outside of the main lib and then it could fall into one of the nice scenarios.

Reply:
Sounds reasonable, I guess we can leave it there as a reminder.